### PR TITLE
Adding docs to gcore and minor refactoring of syscalls

### PR DESCRIPTION
**Pull request by peter-cmw**
Added initial documentation draft to gcore modules
Remove send methods that do not send value
Return Message Id  from reply method

### CODE REVIEW

File: gcore/src/exec.rs [hunk 1]
```diff
@@ -16,6 +16,10 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+//! Sys calls related to smart contract execution.
```
Comment (NikVolf) [inline gcore/src/exec.rs:19]:
```suggestion
//! Sys calls related to program execution flow.
```

**Comment by NikVolf** [comment]
We should avoid notion of "smart contracts"

Those are "programs"

(there are multiple issues about that in the pr)

File: gcore/src/exec.rs [hunk 2]
```diff
@@ -16,6 +16,10 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+//! Sys calls related to program execution flow.
+//!
+//! Provides api for low-level async implementation (probably gas should be other mod)
```
Comment (NikVolf) [inline gcore/src/exec.rs:21]:
```suggestion
//! Provides api for low-level async implementation
```

**Review by NikVolf — APPROVED** [review]

File: gcore/src/exec.rs [hunk 3]
```diff
@@ -26,14 +30,69 @@ mod sys {
...
+///
+/// Each execution of message processing consumes gas, both on instructions and memory allocations.
+/// gas_available() returns value of the gas available for spend during current execution.
+///
+/// # Examples
+///
+/// ```
+/// use gcore::{msg, exec};
+///
+/// // Perform work while gas_available is more then 1000
```
Comment (shamilsan) [inline gcore/src/exec.rs:43]:
```suggestion
/// // Perform work while gas_available is more than 1000
```

**Review by shamilsan — CHANGES_REQUESTED** [review]
Please don't merge now, I'll fix typos and merge it.

**Pull request merged**